src/monattack.cpp

```diff
@@ -4647,3 +4648,19 @@ bool mattack::dodge_check( monster *z, Creature *target ) {
     }
     return false;
 }
+
+void mattack::lay_egg( monster *z )
+{
+    monster &critter = *z;
+    const furn_t &furn = g->m.furn( critter.pos() ).obj();
+
+    if( !critter.is_dead() && critter.has_flag( MF_EGGLAYING ) && furn.has_flag( TFLAG_NEST ) &&
```

Comment:
Made `lay_egg()` mattack, and it works if I spawn a chicken on a nest, but now for some reason it seems that the chickens aren't interested in going to the nest. Gonna see why and how to fix it.